osu.Game/Database/LegacyBeatmapImporter.cs

```diff
@@ -22,22 +22,42 @@ protected override IEnumerable<string> GetStableImportPaths(Storage storage)
         {
             // make sure the directory exists
             if (!storage.ExistsDirectory(string.Empty))
-                yield break;
+                return Array.Empty<string>();
 
-            foreach (string directory in storage.GetDirectories(string.Empty))
-            {
-                var directoryStorage = storage.GetStorageForDirectory(directory);
+            List<string> paths = new List<string>();
 
-                if (!directoryStorage.GetFiles(string.Empty).ExcludeSystemFileNames().Any())
+            try
+            {
+                foreach (string directory in storage.GetDirectories(string.Empty))
                 {
-                    // if a directory doesn't contain files, attempt looking for beatmaps inside of that directory.
-                    // this is a special behaviour in stable for beatmaps only, see https://github.com/ppy/osu/issues/18615.
-                    foreach (string subDirectory in GetStableImportPaths(directoryStorage))
-                        yield return subDirectory;
+                    var directoryStorage = storage.GetStorageForDirectory(directory);
+
+                    try
+                    {
+                        if (!directoryStorage.GetFiles(string.Empty).ExcludeSystemFileNames().Any())
+                        {
+                            // if a directory doesn't contain files, attempt looking for beatmaps inside of that directory.
+                            // this is a special behaviour in stable for beatmaps only, see https://github.com/ppy/osu/issues/18615.
+                            foreach (string subDirectory in GetStableImportPaths(directoryStorage))
+                                paths.Add(subDirectory);
+                        }
+                        else
+                            paths.Add(storage.GetFullPath(directory));
+                    }
+                    catch (Exception e)
+                    {
+                        // Catch any errors when enumerating files
+                        Logger.Log($"Error when enumerating files in {directoryStorage.GetFullPath(string.Empty)}: {e}");
```

Comment:
Hmm, shouldn't this be marked with "error" level verbosity? Would imagine it notifying the user about the error so that they're not confused why none of their beatmaps are imported, but feeling like I'm missing something.

Reply:
I'm not sure. It will show the count as zero in the first run dialog, and if they think this is a problem they could provide logs to us which will help us remedy the situation.